src/jdk.management/windows/native/libmanagement_ext/OperatingSystemImpl.c

```diff
@@ -775,8 +777,7 @@ static int
 getCurrentQueryIndexForProcess() {
     int currentQueryIndex = currentQueryIndexForProcess();
 
-    assert(currentQueryIndex >= 0 &&
-           currentQueryIndex < numberOfJavaProcessesAtInitialization);
+    assert(currentQueryIndex < numberOfJavaProcessesAtInitialization);
```

Comment:
doesn't it make sense to change assert to currentQueryIndex >= -1, so any other negative numbers are still hit assertion?

Reply:
Thanks Leonid, I didn't see much value in that - we can see in the same file the return value of currentQueryIndexForProcess() is going to be -1 or an index from the for loop iteration from 0 to INT_MAX.
We can do that if you really feel it has a benefit?

Reply:
What was the cause of all the 0's in the output, and how did you get rid of them so you could see the assert message?

Reply:
Thanks Leonid!

On the \u0000 content, I don't see that mess in the console (cygwin) if I make these asserts trigger in a local windows build and test.
But I do see the \u0000... in the .jtr file created by jtreg.  So a character encoding disagreement when the output is captured and saved..?  The jtreg faq mentions this, https://openjdk.org/jtreg/faq.html but I haven't worked on it further.

In logs/bug reports that have them, I just search and replace to remove them...

Reply:
Thanks Chris!